quickwit/quickwit-indexing/src/actors/merge_executor.rs

```diff
@@ -386,8 +386,9 @@ impl MergeExecutor {
             "Delete tasks can be applied only on one split."
         );
         assert_eq!(tantivy_dirs.len(), 1);
-        let split = splits.pop().unwrap();
+        let split = &splits[0];
         let index_metadata = self.metastore.index_metadata(&split.index_id).await?;
+        ctx.record_progress();
```

Comment:
*mind trick*

there is no `record_progress` here.

Reply:
you probably want to protect_future the call above instead.